### DISCUSSION TITLE
Intel GPU support

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#232] [General]
Hey Nagadomi, just was curious will there be any plans to eventually support intel gpus?

**Comment by nagadomi** [comment]
Same as https://github.com/nagadomi/nunif/discussions/91 ?

The recently released pytorch 2.4 can easily support Intel GPU on the software side(iw3).
https://pytorch.org/docs/stable/notes/get_start_xpu.html
However, it only supports Linux and WSL2, and requires building from sourcecode, so it is not yet possible to support it on Windows (without WSL).
If binary packages for Windows will be distributed in the future, it can be supported in the same way as MPS(GPU on macOS).

**Reply by 25879** [reply]
> Same as #91 ?
> 
> The recently released pytorch 2.4 can easily support Intel GPU on the software side(iw3). https://pytorch.org/docs/stable/notes/get_start_xpu.html However, it only supports Linux and WSL2, and requires building from sourcecode, so it is not yet possible to support it on Windows (without WSL). If binary packages for Windows will be distributed in the future, it can be supported in the same way as MPS(GPU on macOS).

Pytorch 2.5 have been released, which support Win 11/10
[https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpu/2-5.html], 
I tried to add "torch.xpu.is_available()" to the code but not working... any suggestion?

**Reply by nagadomi** [reply]
xpu is still preview. You need to install test/xpu binary wheel.
See https://pytorch.org/docs/stable/notes/get_start_xpu.html

Also I don't have Intel GPU and no env to try it out, so I don't know if it will work or not.

**Reply by nagadomi** [reply]
I found that Intel GPU can be tested in Intel Tiber AI Cloud, so I will try to support it.

**Reply by kaelsonofkrypto** [reply]
thats so awesome of you to follow up on trying to support this brother, thanks so much. this is probably the best news i have heard all day. thanks again

**Reply by nagadomi** [reply]
On Intel Tiber AI Cloud (driver, oneAPI etc preinstalled) iw3 is already working with `xpu` branch code.
Intel Max 1100 can achieve 20-30FPS for 1080p video with Any_S.
Unlike MPS, it seems to be able to achieve similar performance as NVIDIA card.

I'm still not clear on the environment setup, but at least the bat files need to be updated for xpu.
